File: libJava/com/earndero/sqlwrapper/java/Statement.java
package com.earndero.sqlwrapper.java;

import java.sql.SQLException;

public class Statement {
    java.sql.Statement statement;
    //constructor visible only in package
    Statement(){};

    public void exec(String s) {
        try {
            statement.executeUpdate(s);
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
    }

    public Cursor query(String tableNsme, String[] names) {
        java.sql.ResultSet rs = null;
        StringBuilder stmt = new StringBuilder("SELECT ");
        for (int i=0; i<names.length; i++) {
            if (i>0) stmt.append(",");
            stmt.append(names[i]);
        }
        stmt.append(" FROM ").append(tableNsme);
        try {
            rs = statement.executeQuery(stmt.toString());
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
        return new Cursor(rs);
    }
}

File: libJavaKt/com/earndero/sqlwrapper/javakt/Statement.kt
package com.earndero.sqlwrapper.javakt

import java.sql.ResultSet
import java.lang.StringBuilder
import java.sql.SQLException
import java.sql.Statement

class Statement  //constructor visible only in package
internal constructor() {
    var statement: Statement? = null
    fun exec(s: String?) {
        try {
            statement!!.executeUpdate(s)
        } catch (throwables: SQLException) {
            throwables.printStackTrace()
        }
    }

    fun query(tableNsme: String?, names: Array<String?>): Cursor {
        var rs: ResultSet? = null
        val stmt = StringBuilder("SELECT ")
        for (i in names.indices) {
            if (i > 0) stmt.append(",")
            stmt.append(names[i])
        }
        stmt.append(" FROM ").append(tableNsme)
        try {
            rs = statement!!.executeQuery(stmt.toString())
        } catch (throwables: SQLException) {
            throwables.printStackTrace()
        }
        return Cursor(rs)
    }
}